### ISSUE TITLE
Incorrect error message displayed when uploading an invalid swagger file

### ISSUE THREAD

**Issue by gashcrumb** [syndesisio/syndesis#2315]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

When uploading an invalid swagger file it appears the API can either return an HTTP 200 with an `error` attribute *or* return an HTTP 500 error response with the following:

```
{"errorCode":500,"userMsg":"Please contact the administrator and file a bug report","developerMsg":"Internal Server Exception. Swagger specification does not provide a `schemes` definition and the Swagger specification was uploaded so the originating URL is lost to determine the scheme to use"}
```

In the latter case where the above response is received the wrong error message is shown in the UI:

<img width="781" alt="screenshot 2018-04-10 09 52 58" src="https://user-images.githubusercontent.com/351660/38561050-0640be34-3ca5-11e8-808d-fae7c7a14359.png">

Probably caught by a generic error handler and not correctly propagated back to the component.

This can be reproduced by trying to create an API client connector with this swagger file:

[kie-server-swagger.json.txt](https://github.com/syndesisio/syndesis/files/1894902/kie-server-swagger.json.txt)

@chirino FYI

**Comment by zregvart** [comment]
Parts of this should be addressed with #2310. I'll leave this open so we can see if we can improve this more.

**Comment by gashcrumb** [comment]
Sounds good.  The main problem I think here though is that on the UI side the HTTP 500 response is handled *somewhere* in the stack underneath the component, and it's apparently attempting to parse something in the response as JSON, am not sure if that's valid.  Had a quick look yesterday and found that a breakpoint in the error handler didn't get hit when the response comes back, so it seems to hit this in the angular http client service :-/

**Comment by gashcrumb** [comment]
Found it, however @zregvart for this case can we just use the value given in `developerMsg` for the `userMsg` attribute in the error response?  Otherwise when I fix this in the UI we get:

<img width="769" alt="screenshot 2018-04-11 09 43 01" src="https://user-images.githubusercontent.com/351660/38620570-bfba12a8-3d6c-11e8-8a26-5dd541d765f1.png">

**Comment by deeleman** [comment]
> it's apparently attempting to parse something in the response as JSON, am not sure if that's valid

The reason why this was implemented this way is because in some case scenarios, the error is returned as a string. Yeah, you read it right: A string containing an object literal. That made debugging API errors pretty much impossible.

**Comment by gashcrumb** [comment]
Yep, I think though we should take that out (linked PR) and insist on fixing those kinds of responses from this point on instead, as otherwise it's a nightmare to deal with :-)

**Comment by zregvart** [comment]
> Found it, however @zregvart for this case can we just use the value given in developerMsg for the userMsg attribute in the error response? Otherwise when I fix this in the UI we get:

I'm not sure if `developerMsg` will contain something that we want the end user to see. Might be something like `"NullPointerException: null"`.

Would setting `userMsg` on server side in all cases to something like `"Please contact the administrator and file a bug report"` if no `userMsg` is specified by the throwing code help?

**Comment by gashcrumb** [comment]
So what's shown in the UI is whatever `userMsg` contains, in this case I think we should show swagger validation errors and not the generic message.  Even if the user doesn't understand what the error is they can still grab a screenshot to ask someone about it.

I only mention this as if you look at the example error response in the description, `developerMsg` contains something meaningful, while `userMsg` does not.

**Comment by zregvart** [comment]
> I only mention this as if you look at the example error response in the description, developerMsg contains something meaningful, while userMsg does not.

Though with #2307 this will no longer be the case, now we don't throw an exception from parsing/generating code but instead return this as a part of warnings/errors properties of the `POST /api/v1/connectors/custom/info` response.

**Comment by gashcrumb** [comment]
ok, perfect!  Thanks!
